02-oop-inheritance/jsES5/EventEmitter.js

```diff
@@ -0,0 +1,66 @@
+//EventEmitter class
+class EventEmitter{
+    constructor(){
+        this.event = {};
+    }
+
+    //EventPlay = Event added - callback = function added
+    on(eventPlay, callback){  
+        //If the event exists within the object
+        if(this.event[eventPlay]){
+                //Evaluate if it is associated with the same function
+                if(this.event[eventPlay].indexOf(callback) != -1){
+                    console.log('Its already exists');
+                }
+                //If not, I assigned the new function
+                else{
+                    this.event[eventPlay].push(callback);
+                }
+        }          
+        else{
+            //If not exist, i create the event as an array, and i assigned the function
+            this.event[eventPlay] = []
+            this.event[eventPlay].push(callback);
+        }      
+    }
+
+    //Receive the event to be deleted with the function
+    off(eventPlay, callback){
+        //If the event exist and there are more than one
+        if(this.event[eventPlay].length > 1){ 
```

Comment:
If the event is `undefined` this will throw an error because Javascript couldn't read the property length of `undefined`